front/src/components/Search/Result/ResultProfileDrawer.tsx

```diff
@@ -27,7 +28,7 @@ const ResultProfileDrawer: React.FC<IResultProfileDrawerProps> = ({
         resultProfile!.id,
         EProfileType.HOST === profileType
       ).then((res) => {
-        setProfile(res?.data);
+        setProfile(buildProfile(res?.data as IHostProfile) as IHostProfile);
```

Comment:
Is it me or is buildProfile() not taking any parameters ?

Reply:
My bad, GitHub was redirecting to a method with the same name

Reply:
But why do we cast res.data to IHostPtofile when buildProfile takes *any*. I think isntad of *any* you should have IHostProfile | IArtistProfile, so no need to cast it afterwards